src/function/special/riemannZeta.js

```diff
@@ -0,0 +1,87 @@
+import { factory } from '../../utils/factory.js'
+
+const name = 'riemannZeta'
+const dependencies = ['typed', 'config', 'multiply', 'pow', 'divide', 'factorial', 'gamma', 'sin', 'subtract', 'add', 'Complex']
+
+export const createRiemannZeta = /* #__PURE__ */ factory(name, dependencies, ({ typed, config, multiply, pow, divide, factorial, gamma, sin, subtract, add, Complex }) => {
+  /**
+   * Compute the Riemann Zeta function of a value using an infinite series for
```

Comment:
The current implementation is really elegant and concise. 

In an earlier implementation you where using [Chebychev coefficients](https://github.com/josdejong/mathjs/issues/2904#issuecomment-1483912918). Does the algorithm you implemented now have any drawbacks that we need to be aware of?

Reply:
Thanks for the clarification. And indeed we can further tune the function in the future if needed.